applications/ray/kuberay-tpu-webhook/main.go

```diff
@@ -340,10 +347,11 @@ func getReplicaIndex(clusterName string, namespace string) int {
 			}
 		}
 	}
+	// first pod of new slice in cluster
 	if nextLowestId == math.MaxInt32 {
-		klog.ErrorS(errors.New("Replica Index never set"), "RayCluster", namespace+"/"+clusterName, "Replica Index", nextLowestId)
+		nextLowestId = numSlices
```

Comment:
When would we get here? If `sliceToWorkers` is populated when we validate the ray cluster, then shouldn't we always find a match?

Reply:
`sliceToWorkers` isn't populated when we validate the RayCluster, `sliceToHostnames` is. We populate `sliceToWorkers` as `TPU_WORKER_ID`s are assigned in `getNextWorkerID`. We get to this if statement when  `sliceToWorkers` isn't empty (so some slices / worker groups have already been added) and we intercept the first pod in a different worker group.